From 2ca17076097f3c5272a619b0f0195cb8ecca4c01 Mon Sep 17 00:00:00 2001
From: David Gibson <dgibson@redhat.com>
Date: Mon, 15 Sep 2014 07:12:18 -0400
Subject: [virt] kvm/ppc: book3s/pr - Fix compilation without CONFIG_ALTIVEC

Message-id: <1410765214-16377-3-git-send-email-dgibson@redhat.com>
Patchwork-id: 94701
O-Subject: [PATCH 02/78] KVM: PPC: Book3S PR: Fix compilation without CONFIG_ALTIVEC
Bugzilla: 1123145 1123133 1123367
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1123145

Prerequisite to backport 99dae3bad28d8fdd32b7bfdd5e2ec7bb2d4d019d

Signed-off-by: David Gibson <dgibson@redhat.com>

commit f24817716efe52f3c8fc137c6893963a0416f3cb
Author: Paul Mackerras <paulus@samba.org>
Date:   Fri Sep 20 14:52:42 2013 +1000

    KVM: PPC: Book3S PR: Fix compilation without CONFIG_ALTIVEC

    Commit 9d1ffdd8f3 ("KVM: PPC: Book3S PR: Don't corrupt guest state
    when kernel uses VMX") added a call to kvmppc_load_up_altivec() that
    isn't guarded by CONFIG_ALTIVEC, causing a link failure when building
    a kernel without CONFIG_ALTIVEC set.  This adds an #ifdef to fix this.

    Signed-off-by: Paul Mackerras <paulus@samba.org>
    Signed-off-by: Alexander Graf <agraf@suse.de>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

Conflicts:
 arch/powerpc/kvm/book3s_pr.c
---

diff --git a/arch/powerpc/kvm/book3s_pr.c b/arch/powerpc/kvm/book3s_pr.c
index 425caf5..9bc43fc 100644
--- a/arch/powerpc/kvm/book3s_pr.c
+++ b/arch/powerpc/kvm/book3s_pr.c
@@ -698,10 +698,12 @@ static void kvmppc_handle_lost_ext(struct kvm_vcpu *vcpu)
   enable_kernel_fp();
   load_fp_state(&current->thread.fp_state);
  }
+#ifdef CONFIG_ALTIVEC
  if (lost_ext & MSR_VEC) {
   enable_kernel_altivec();
   load_vr_state(&current->thread.vr_state);
  }
+#endif
  current->thread.regs->msr |= lost_ext;
 }
 
-- 
1.7.1